packages/catalog-client/src/CatalogClient.ts

```diff
@@ -49,6 +50,9 @@ import type {
   AnalyzeLocationResponse,
 } from '@backstage/plugin-catalog-common';
 
+// Number of entities to return in a single streamEntities request
+const STREAM_ENTITIES_LIMIT = 500;
```

Comment:
@freben can we suggest any "magic number" here? I remember we ran some benchmarks internally

Reply:
Hm I don't think I have the numbers at hand. This seems like a reasonable compromise number!

Reply:
I added the `batchSize` parameter to the request but kept the default as 500